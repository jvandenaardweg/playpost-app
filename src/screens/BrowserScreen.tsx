// @ts-ignore
import React from 'react';
import { View, Animated } from 'react-native';
import { NavigationScreenProp, NavigationRoute, NavigationStackScreenOptions } from 'react-navigation';
import { WebView } from 'react-native-webview';

import { ButtonReload } from '../components/ButtonReload';

import colors from '../constants/colors';

interface Props {
  navigation: NavigationScreenProp<NavigationRoute>;
}

interface State {
  loadProgress: string;
}

export class BrowserScreen extends React.PureComponent<Props, State> {

  /* tslint:disable-next-line no-any */
  private webviewRef = React.createRef<any>();

  static navigationOptions = ({ navigation }: { navigation: NavigationScreenProp<NavigationRoute> }): NavigationStackScreenOptions => {
    return {
      title: navigation.getParam('title', null),
      headerRight: <ButtonReload onPress={navigation.getParam('handleOnReload')} />
    };
  }

  animatedWidth = new Animated.Value(0);

  interpolatedWidth = this.animatedWidth.interpolate({
    inputRange: [0, 1],
    outputRange: ['0%', '100%']
  });

  state = {
    loadProgress: ''
  };

  componentDidMount() {
    this.props.navigation.setParams({ handleOnReload: this.handleOnReload });
  }

  handleOnReload = () => {
    this.webviewRef.current && this.webviewRef.current.reload();
  }

  handleOnLoadEnd = () => {
    this.animatedWidth.setValue(0);
  }

  // TODO: fix no any
  /* tslint:disable no-any */
  handleOnLoadProgress = (event: any) => {
    Animated.timing(
      this.animatedWidth,
      {
        toValue: event.nativeEvent.progress,
        duration: 300
      }
    ).start();
  }

  render() {
    const url = this.props.navigation.getParam('url', null);

    return (
      <View style={{ flex: 1, backgroundColor: colors.appBackground }}>
        <Animated.View style={{ position: 'absolute', top: 0, left: 0, width: this.interpolatedWidth, backgroundColor: colors.tintColor, height: 5 }}></Animated.View>
        <WebView
          ref={this.webviewRef}
          source={{ uri: url }}
          startInLoadingState={true}
          // onLoadProgress={this.handleOnLoadProgress}
          // onLoadEnd={this.handleOnLoadEnd}
          sharedCookiesEnabled={true}
          thirdPartyCookiesEnabled={true}
          incognito={false}
          decelerationRate="normal"
        />
      </View>

    );
  }
}
