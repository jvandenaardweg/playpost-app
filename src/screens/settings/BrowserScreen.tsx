// @ts-ignore
import React from 'react';
import { Animated, View } from 'react-native';
import { WebView } from 'react-native-webview';
import { NavigationRoute, NavigationScreenProp, NavigationStackScreenOptions } from 'react-navigation';

import { ButtonReload } from '../../components/ButtonReload';

import colors from '../../constants/colors';

interface Props {
  navigation: NavigationScreenProp<NavigationRoute>;
}

interface State {
  loadProgress: string;
}

export class BrowserScreen extends React.PureComponent<Props, State> {

  public static navigationOptions = ({ navigation }: { navigation: NavigationScreenProp<NavigationRoute> }): NavigationStackScreenOptions => {
    return {
      title: navigation.getParam('title', null),
      headerRight: <ButtonReload onPress={navigation.getParam('handleOnReload')} />
    };
  }

  public animatedWidth = new Animated.Value(0);

  public interpolatedWidth = this.animatedWidth.interpolate({
    inputRange: [0, 1],
    outputRange: ['0%', '100%']
  });

  public state = {
    loadProgress: ''
  };

  /* tslint:disable-next-line no-any */
  private webviewRef = React.createRef<any>();

  public componentDidMount() {
    this.props.navigation.setParams({ handleOnReload: this.handleOnReload });
  }

  public handleOnReload = () => this.webviewRef.current && this.webviewRef.current.reload()

  public handleOnLoadEnd = () => {
    this.animatedWidth.setValue(0);
  }

  // TODO: fix no any
  /* tslint:disable no-any */
  public handleOnLoadProgress = (event: any) => {
    Animated.timing(
      this.animatedWidth,
      {
        toValue: event.nativeEvent.progress,
        duration: 300
      }
    ).start();
  }

  public render() {
    const url = this.props.navigation.getParam('url', null);

    return (
      <View style={{ flex: 1, backgroundColor: colors.appBackground }}>
        <Animated.View style={{ position: 'absolute', top: 0, left: 0, width: this.interpolatedWidth, backgroundColor: colors.tintColor, height: 5 }}></Animated.View>
        <WebView
          ref={this.webviewRef}
          source={{ uri: url }}
          startInLoadingState={true}
          // onLoadProgress={this.handleOnLoadProgress}
          // onLoadEnd={this.handleOnLoadEnd}
          sharedCookiesEnabled={true}
          thirdPartyCookiesEnabled={true}
          incognito={false}
          decelerationRate="normal"
        />
      </View>

    );
  }
}
