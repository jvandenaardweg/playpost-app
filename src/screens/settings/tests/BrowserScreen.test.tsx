import React from 'react';
import { render, RenderAPI } from 'react-native-testing-library';
import { BrowserScreen } from '../BrowserScreen';

const navigateHandler = jest.fn();
const navigationGetParamHandler = jest.fn();
const navigationGoBackHandler = jest.fn();
const navigationSetParamsHandler = jest.fn();

const props: any = {
  navigation: {
    navigate: navigateHandler,
    getParam: navigationGetParamHandler,
    goBack: navigationGoBackHandler,
    setParams: navigationSetParamsHandler,
  },
};

describe('BrowserScreen', () => {

  describe('rendering', () => {
    let wrapper: RenderAPI;

    beforeAll(() => {
      wrapper = render(<BrowserScreen {...props} />);
    });

    it('should render correctly', () => {
      expect(wrapper.toJSON()).toMatchSnapshot();
    });
  });
});
