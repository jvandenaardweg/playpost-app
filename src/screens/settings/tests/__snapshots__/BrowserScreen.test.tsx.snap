// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrowserScreen rendering should render correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#EEEEEE",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#037DE2",
        "height": 5,
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": "0%",
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
        "overflow": "hidden",
      }
    }
  >
    <RNCWKWebView
      cacheEnabled={true}
      decelerationRate={0.998}
      incognito={false}
      messagingEnabled={false}
      onLoadingError={[Function]}
      onLoadingFinish={[Function]}
      onLoadingProgress={[Function]}
      onLoadingStart={[Function]}
      onMessage={[Function]}
      onShouldStartLoadWithRequest={[Function]}
      sharedCookiesEnabled={true}
      source={
        Object {
          "uri": undefined,
        }
      }
      startInLoadingState={true}
      style={
        Array [
          Object {
            "flex": 1,
            "overflow": "hidden",
          },
          Object {
            "backgroundColor": "#ffffff",
          },
          undefined,
          Object {
            "flex": 0,
            "height": 0,
          },
        ]
      }
      thirdPartyCookiesEnabled={true}
      useSharedProcessPool={true}
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <ActivityIndicator
        animating={true}
        color="#999999"
        hidesWhenStopped={true}
        size="small"
      />
    </View>
  </View>
</View>
`;
